### PR TITLE
DEVX-244: fix serdes in ValidationsAggregatorService

### PR DESCRIPTION
**Pull request by ybyzek**
Issue #150

### CODE REVIEW

**Comment by mjsax** [comment]
Cannot follow (I did read the ticket and the other PRs): what is the issue with those `Serdes` being `private final`? Can you elaborate?

**Comment by vvcephei** [comment]
@mjsax @ybyzek Just for curiosity's sake, what was the reason? I also couldn't find it in the ticket.

**Comment by ybyzek** [comment]
@vvcephei , @benstopford gave a much clearer explanation in #138 .  (I may have mis-spoken about the `final` keyword)

**Comment by vvcephei** [comment]
Yikes! That's a tricky dependency.

I see, so the actual problem was just that the serdes were created during class construction, when they actually must be preceeded by a call to `io.confluent.examples.streams.microservices.util.MicroserviceUtils#parseArgsAndConfigure`

Then it should be ok to make them `private final` local variables.

Another way to accomplish this would be to change only the `main` method:
```java
  public static void main(final String[] args) throws Exception {
    final String bootstrapServers = parseArgsAndConfigure(args);
    final ValidationsAggregatorService service = new ValidationsAggregatorService();
    service.start(bootstrapServers, "/tmp/kafka-streams");
    addShutdownHookAndBlock(service);
  }
```

It should also be possible to set a flag in `Schemas` to ensure that no one can get the topics until after the call to `io.confluent.examples.streams.microservices.domain.Schemas#configureSerdesWithSchemaRegistryUrl`

**Comment by ybyzek** [comment]
@vvcephei -- just to be clear, are you ok with the PR as it is right now?  Or is there a specific recommendation you are making?

**Comment by benstopford** [comment]
Both approaches should be fine, although @vvcephei's suggestions seem clearer. LGTM

File: src/main/java/io/confluent/examples/streams/microservices/ValidationsAggregatorService.java [hunk 1]
```diff
@@ -68,6 +54,21 @@ private KafkaStreams aggregateOrderValidations(
       final String stateDir) {
     final int numberOfRules = 3; //TODO put into a KTable to make dynamically configurable
 
+    Consumed<String, OrderValidation> serdes1 = Consumed
```
Comment (bbejeck) [inline src/main/java/io/confluent/examples/streams/microservices/ValidationsAggregatorService.java]: needs to be `final` here and below to `serdes7`

**Comment by ybyzek** [comment]
I have recoded to use @vvcephei 's suggestion.  Please review updated code changes (much cleaner).

**Comment by vvcephei** [comment]
Thanks, @ybyzek ! LGTM.

(sorry, I didn't see your earlier question until now)

**Pull request merged**